application/src/main/java/org/thingsboard/server/controller/AlarmController.java

```diff
@@ -179,6 +188,51 @@ public void clearAlarm(@ApiParam(value = ALARM_ID_PARAM_DESCRIPTION) @PathVariab
         tbAlarmService.clear(alarm, getCurrentUser()).get();
     }
 
+    @ApiOperation(value = "Assign/Reassign Alarm (assignAlarm)",
+            notes = "Assign the Alarm. " +
+                    "Once assigned, the 'assign_ts' field will be set to current timestamp and special rule chain event 'ALARM_ASSIGNED' " +
+                    "(or ALARM_REASSIGNED in case of assigning already assigned alarm) will be generated. " +
+                    "Referencing non-existing Alarm Id will cause an error." + TENANT_OR_CUSTOMER_AUTHORITY_PARAGRAPH, produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/alarm/{alarmId}/assign/{assigneeId}", method = RequestMethod.POST)
+    @ResponseStatus(value = HttpStatus.OK)
+    public void assignAlarm(@ApiParam(value = ALARM_ID_PARAM_DESCRIPTION)
+                            @PathVariable(ALARM_ID) String strAlarmId,
+                            @ApiParam(value = ASSIGN_ID_PARAM_DESCRIPTION)
+                            @PathVariable(ASSIGNEE_ID) String strAssigneeId
+    ) throws Exception {
+        checkParameter(ALARM_ID, strAlarmId);
+        checkParameter(ASSIGNEE_ID, strAssigneeId);
+        AlarmId alarmId = new AlarmId(toUUID(strAlarmId));
+        UserId assigneeId = new UserId(UUID.fromString(strAssigneeId));
+        Alarm alarm = checkAlarmId(alarmId, Operation.WRITE);
+        User assigneeUser = userService.findUserById(getTenantId(), assigneeId);
+
+        SecurityUser assigneeSecurityUser = new SecurityUser(assigneeUser, false, null);
+        try {
+            checkEntityId(alarm.getOriginator(), Operation.WRITE, assigneeSecurityUser);
```

Comment:
This check is redundant. We just need to check that the current user has access to the alarm and the assignee belongs to the same tenant.